src/imitation/rewards/reward_nets.py

```diff
@@ -221,58 +137,141 @@ def __init__(
             # Otherwise, it can be arbitrary, so make a trainable variable.
             self.end_potential = nn.Parameter(th.zeros(()))
 
-    @property
-    @abc.abstractmethod
-    def potential_net(self) -> nn.Module:
-        """The reward shaping network (disentangles dynamics from reward).
-
-        Returned `nn.Module` should map batches of observations to batches of
-        scalar potential values."""
-
-    def reward_train(
+    def forward(
         self,
         state: th.Tensor,
         action: th.Tensor,
         next_state: th.Tensor,
         done: th.Tensor,
-    ) -> th.Tensor:
-        """Compute the (shaped) training reward of each timestep."""
-        base_reward_net_output = self.base_reward_net(state, action, next_state, done)
-        new_shaping_output = self.potential_net(next_state).flatten()
-        old_shaping_output = self.potential_net(state).flatten()
+    ):
+        base_reward_net_output = self.base(state, action, next_state, done)
+        new_shaping_output = self.potential(next_state).flatten()
+        old_shaping_output = self.potential(state).flatten()
         done_f = done.float()
         new_shaping = done_f * self.end_potential + (1 - done_f) * new_shaping_output
         final_rew = (
             base_reward_net_output
-            + self._discount_factor * new_shaping
+            + self.discount_factor * new_shaping
             - old_shaping_output
         )
         assert final_rew.shape == state.shape[:1]
         return final_rew
 
+
+class AIRLRewardNet(nn.Module):
+    """Wrapper around RewardNet with different forward passes for test and train.
+
+    This class is used for AIRL, where we want the potential shaping to be included
+    during training but not testing. If the wrapped RewardNet is not an instance
+    of ShapedRewardNet, then the test and train passes will be identical.
+
+    TODO(ejnnr): get rid of this class entirely, instead just implement the logic
```

Comment:
Yeah I agree this doesn't belong here. Not sure if it's better to just move the class to AIRL-specific code or delete it entirely.

Reply:
I've removed it entirely now, I think that makes the most sense. It's significantly shorter in terms of LOC and it also "flattens" things a bit because we don't have this wrapper everywhere.